### PR TITLE
Make vaultkeeper wait for child process and terminate when process te…

### PR DESCRIPTION
**Pull request by mracter**
…rminates, rather than using a static sleep interval.

### CODE REVIEW

File: vaultkeeper/vaultkeeper.py [hunk 1]
```diff
@@ -129,9 +129,14 @@ def run(self):
                                shell=False
                                )
         while app.poll() is None:
-            self.renew_token()
-            self.renew_all()
-            time.sleep(self.configs.refresh_interval)
+            try:
+                app.wait(timeout=self.configs.refresh_interval)
+            except TimeoutExpired as tex:
```
Comment (jerith) [inline vaultkeeper/vaultkeeper.py]: This probably needs to be `subprocess.TimeoutExpired`.

File: vaultkeeper/vaultkeeper.py [hunk 2]
```diff
@@ -129,9 +129,14 @@ def run(self):
...
+                self.logger.info('Renewing leases...')
+                self.renew_token()
+                self.renew_all()
+            else:
+                system.exit(0)
```
Comment (jerith) [inline vaultkeeper/vaultkeeper.py]: I'm not entirely comfortable with `sys.exit()` in a method like this. Maybe we should `return app.returncode` and then pass that to `sys.exit()` in `main()`?

Comment (mracter) [reply vaultkeeper/vaultkeeper.py]: Makes sense - I'll do that rather.

**Review by jerith — APPROVED** [review]
:+1:

**Pull request merged**